src/layout/IHyprLayout.cpp

```diff
@@ -178,6 +182,54 @@ void IHyprLayout::onWindowCreatedFloating(PHLWINDOW pWindow) {
     }
 }
 
+bool IHyprLayout::onWindowCreatedAutoGroup(PHLWINDOW pWindow) {
+    static auto AUTOGROUP = CConfigValue<Hyprlang::INT>("group:auto_group");
+    if ((*AUTOGROUP || g_pInputManager->m_bWasDraggingWindow) && g_pCompositor->m_pLastWindow.lock() && g_pCompositor->m_pLastWindow != pWindow &&
+        g_pCompositor->m_pLastWindow->m_pWorkspace == pWindow->m_pWorkspace && g_pCompositor->m_pLastWindow->m_sGroupData.pNextWindow.lock() &&
+        pWindow->canBeGroupedInto(g_pCompositor->m_pLastWindow.lock()) && !g_pXWaylandManager->shouldBeFloated(pWindow)) {
+
+        switch (pWindow->m_bIsFloating) {
```

Comment:
what is this cursed switch

Reply:
in this situation I think that `switch...case` makes the code easier to read than `if...else`, but tell me what you prefer and I'll change it.